### PR TITLE
Keep the stores open all the time

### PR DESCRIPTION
**Pull request by badboy**
A "store" is a handle to the database.
Glean has 3 different ones and can keep them open throughout its lifetime.

Not frequently re-opening database handles also might be a performance
improvement.

Extracted from #664 because it's unrelated. Not directly related to a filed bug.

### CODE REVIEW

**Comment by badboy** [comment]
We didn't know at submission time, it was mostly a hunch from Dexter.
I did ran a very simplistic benchmark (setting 2 metrics and submitting a ping) and get reliable 5% improvements compared to what's in current master.

```
glean set and submit    time:   [94.437 us 94.502 us 94.577 us]
                        change: [-5.3865% -5.1006% -4.8247%] (p = 0.00 < 0.05)
                        Performance has improved.
```

That is on my Linux machine (on my Mac it's unreliable and significantly slower, milliseconds instead of microseconds)

I'll work on adding the benchmark as part of this as well.

**Comment by badboy** [comment]
So the numbers are unreliable even on my Linux workstation. The benchmark is also _very_ simplistic and nowhere near a real use case.
Nonetheless I included the benchmark now as an extra crate so we can re-run/extend it if needed.

**Comment by badboy** [comment]
holding back on this to investigate what this means for I/O operations.

**Comment by badboy** [comment]
All good, I/O operations unchanged for the startup case.

/cc @travis79, please give it a final review

File: glean-core/benchmark/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+cargo bench -p benchmark
+```
+
+This is also available as a `make` task:
+
+```
+make bench-rust
+```
+
+Any additional run compare results to the preceding run.
```
Comment (travis79) [inline glean-core/benchmark/README.md]: `compare` -> `compares`

**Pull request merged**